### ISSUE TITLE
Corrupts the OpenGL context of OpenGL-based host UIs

### ISSUE THREAD

**Issue by BeRo1985** [michaelwillis/dragonfly-reverb#73]
The pluigin corrupts the OpenGL context of the host UI on closing the plugin UIs. You should use at least on Windows the following way:

```
  // get previous DC and render context
  hOldDC = wglGetCurrentDC();
  hOldRC = wglGetCurrentContext();

  // get DC for the hwnd of the window to draw in
  hdc = GetDC(hwnd);

  // set rendering context and DC
  wglMakeCurrent(hdc, glRC);

  // do here whatever else you want in the context, for example drawing the UI
  DoYourStuff();

  // clear errors to avoid confusing other code
  // it's fine to call this and ignore errors if you don't
  // care about them; it's just so it reports success on next call
  glGetError();

  // restore previous context and DC
  wglMakeCurrent(hOldDC, hOldRC);

  // release the DC we allocated
  ReleaseDC(hwnd, hdc);
```   

or something in this way, for to avoid that your OpenGL-based plugin UI stuff can corrupt the OpenGL context of the host itself. 

For how this OpenGL context corruption issue looks like, see [https://youtu.be/g-Y6sPm9YYY](url) (which is my own DAW, which I'm developing in the moment)

In summary, make sure you use MakeCurrent() in your context, and assume that the VST host may have its own context between calls to MakeCurrent(). Or, simply do not use OpenGL for the plugin UI if you can not take care of it, that the host itself has also one oder more OpenGL context(s).

At least in my own DAW project, I am paying attention to this, where I myself use wglGetCurrentDC+wglGetCurrentContext and wglMakeCurrent on every OpenGL action block of code. But that doesn't help here, if the plugins don't take it into account in their UIs as well.

**Comment by falkTX** [comment]
This is a fault of DPF, the framework this plugin uses for its UI. (I am the creator of it)
No time at the moment to handle this, but I should have some in November.

**Comment by falkTX** [comment]
This is fixed on the latest builds.

**Comment by michaelwillis** [comment]
I will work on making a new release based on the latest build, hopefully it will be available soon.

**Comment by michaelwillis** [comment]
This is fixed in [release 3.2.6](https://github.com/michaelwillis/dragonfly-reverb/releases/tag/3.2.6).
